radare2 0.10.2

nl5887 opened the pull request:
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Does your submission pass `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Have you built your formula locally prior to submission with `brew install <formula>`?

nl5887 commented:
* bump to 0.10.2
* fixed style errors (Prefer to_s over string interpolation and Avoid comma)
* added test block

ilovezfs commented:
I cannot seem to reproduce the build failure locally.

tdsmith commented:
looks like it will be fixed by https://github.com/Homebrew/brew/pull/63

ilovezfs commented:
Yup that fixed it.
```
==> brew --env
==> brew config
==> brew tests
==> brew tests --no-compat
==> brew readall --aliases --syntax
==> brew uses radare2
==> brew fetch --retry autoconf automake cairo fontconfig freetype gettext git glib gmp gobject-introspection libewf libffi libmagic libpng libtool lua51 makedepend openssl pcre pixman pkg-config swig vala valabind xz yara
==> brew fetch --retry radare2 --build-bottle
==> brew install --only-dependencies --verbose --build-bottle radare2
==> brew install --verbose --build-bottle radare2
==> brew audit radare2
==> brew bottle --verbose --rb radare2
==> brew bottle --merge --write --no-commit ./radare2-0.10.2.el_capitan.bottle.rb
==> brew uninstall --force radare2
==> brew uninstall --force autoconf automake cairo fontconfig freetype gettext git glib gobject-introspection libffi libpng libtool makedepend pixman pkg-config swig vala valabind xz
==> brew install ./radare2-0.10.2.el_capitan.bottle.tar.gz
==> brew test --verbose radare2
==> brew uninstall --force radare2
==> brew uninstall --force gmp libewf libmagic lua51 openssl pcre yara
iMac-TMP:homebrew-core joe$ 

```

apjanke commented:
Homebrew/brew#63 is merged and the subsequent test-bot rerun on this worked. Think you can merge this now.

ilovezfs commented on Formula/radare2.rb:
```diff
@@ -92,4 +93,8 @@ def install
   end
+
+  test do
+    system "#{bin}/r2", "--version"
```
Please have this assert_match against the version.

ilovezfs commented on Formula/radare2.rb:
```diff
@@ -53,7 +53,7 @@ def install
 
     resource("extras").stage do
-      ENV.append_path "PATH", "#{bin}"
+      ENV.append_path "PATH", bin.to_s
```
I don't think we need the `.to_s` at all.

DomT4 replied:
This will be Rubocop's fault _I suspect_. It nags to change it.

ilovezfs commented on Formula/radare2.rb:
```diff
@@ -62,7 +62,7 @@ def install
 
     resource("bindings").stage do
-      ENV.append_path "PATH", "#{bin}"
+      ENV.append_path "PATH", bin.to_s
```
I don't think we need the `.to_s` at all.

ilovezfs commented on Formula/radare2.rb:
```diff
@@ -3,17 +3,17 @@ class Radare2 < Formula
   stable do
-    url "http://radare.org/get/radare2-0.10.1.tar.xz"
-    sha256 "5ac02717786f2ff3b5326927351d5ca38464da89675c8edfb4ded43addb22987"
+    url "http://rada.re/get/radare2-0.10.2.tar.xz"
```
The canonical links still appears to be
http://www.radare.org/get/radare2-0.10.2.tar.xz

ilovezfs commented on Formula/radare2.rb:
```diff
@@ -3,17 +3,17 @@ class Radare2 < Formula
     resource "bindings" do
-      url "http://radare.org/get/radare2-bindings-0.10.1.tar.xz"
-      sha256 "6cb91f4135b5e490185e25629850cb48c08c06882e2c870fa5ab1425cc84106f"
+      url "http://rada.re/get/radare2-bindings-0.10.2.tar.xz"
```
http://www.radare.org/get/radare2-bindings-0.10.2.tar.xz

ilovezfs commented on Formula/radare2.rb:
```diff
@@ -3,17 +3,17 @@ class Radare2 < Formula
     resource "extras" do
-      url "http://radare.org/get/radare2-extras-0.10.1.tar.xz"
-      sha256 "c330210c8e6ce5fa8113c455e98c994fb5ecbb5d2f1c15c40c0d1bcbc24d5092"
+      url "http://rada.re/get/radare2-extras-0.10.2.tar.xz"
```
http://www.radare.org/get/radare2-extras-0.10.2.tar.xz

nl5887 commented:
ok, I've changed the urls to the canonical ones. The reason I used the short urls was because Radare is using them in all of there communication. 

Added assert_match on version, and removed the to_s.

ilovezfs commented on Formula/radare2.rb:
```diff
@@ -92,4 +93,8 @@ def install
   end
+
+  test do
+    assert_match /radare2 0\.10\.2/, shell_output("#{bin}/r2 -version")
```
you can make it `version.to_s` so it doesn't need updating each time.

ilovezfs commented:
other than the `version.to_s` change, :+1:

apjanke commented:
:+1:

tdsmith commented:
Thanks for your pull request! :tada: Homebrew thrives because of contributions like this one.

(Oops, forgot to squash those commits!)
